core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2011,7 +2011,16 @@ class ReplicaManager(val config: KafkaConfig,
           "directory.")
         None
 
-      case HostedPartition.Online(partition) => Some(partition, false)
+      case HostedPartition.Online(partition) => {
+        if (partition.topicId.isDefined && !partition.topicId.equals(topicId)) {
```

Comment:
The second check here is not right since `topicId` is not an option. I think what we want is this?
```scala
if (partition.topicId.exists(_ != topicId)) {
```

Reply:
good catch